src/network/tarcap/packets_handlers/status_packet_handler.cpp

```diff
@@ -66,21 +69,33 @@ void StatusPacketHandler::process(const dev::RLP& packet_rlp, const PacketData&
                    << getFormattedVersion({node_major_version, node_minor_version, node_patch_version})
                    << ", our node version" << TARAXA_VERSION << ", host " << packet_data.from_node_id_.abridged()
                    << " will be disconnected";
-      host->disconnect(packet_data.from_node_id_, dev::p2p::UserReason);
+      if (auto host_shr = host.lock(); host_shr) {
+        host_shr->disconnect(packet_data.from_node_id_, p2p::UserReason);
+      } else {
+        LOG(log_er_) << "Invalid host in StatusPacketHandler";
+      }
```

Comment:
hmm weak_ptr<Host> is part of peers_state class, which is available for every packet handler, let's remove `host` argument from all `process` functions. and use host from `peers_state`